### PR TITLE
fix: prevent race condition in vpc tests

### PR DESCRIPTION
**Pull request by jesterhazy**
*Description of changes:*

fix race condition in vpc tests that causes intermittent test failures

## Merge Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your pull request._

- [x] I have read the [CONTRIBUTING](https://github.com/aws/sagemaker-python-sdk/blob/master/CONTRIBUTING.md) doc
- [x] I used the commit message format described in [CONTRIBUTING](https://github.com/aws/sagemaker-python-sdk/blob/master/CONTRIBUTING.md#commit-message-guidlines)
- [x] I have added tests that prove my fix is effective or that my feature works (if appropriate)
- [x] I have updated any necessary [documentation](https://github.com/aws/sagemaker-python-sdk/blob/master/README.rst) (if appropriate)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: tests/integ/lock.py [hunk 1]
```diff
@@ -15,24 +15,21 @@
...
-import tests.integ
-
-LOCK_PATH = os.path.join(tests.integ.DATA_DIR, 'local_mode_lock')
+DEFAULT_LOCK_PATH = os.path.join(tempfile.gettempdir(), 'sagemaker_test_lock')
 
 
 @contextmanager
-def lock():
-    # Since Local Mode uses the same port for serving, we need a lock in order
-    # to allow concurrent test execution.
```
Comment (laurenyu) [inline tests/integ/lock.py]: I think it'd be good to have an explanation of why these locks are needed somewhere

Comment (jesterhazy) [reply tests/integ/lock.py]: probably at (multiple) points of use. or would a generic "lock for tests that need locks" comment work?

**Comment by jesterhazy** [comment]
i don't need to add to `.gitignore` because the paths are in tempdir outside project. but it sounds like i need to remove an old entry?

File: tests/integ/lock.py (hunk 1 above)
Comment (laurenyu) [reply tests/integ/lock.py]: maybe a generic comment with specific callouts for the most common reasons (e.g. local mode)?

**Comment by jesterhazy** [comment]
### AWS CodeBuild CI Report

* Result: SUCCEEDED
* [Build Logs](https://9dzx6wxcy9.execute-api.us-west-2.amazonaws.com/Prod/buildlogs?key=d434dc3d-ff30-465e-a157-2b3533feb2f3%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Pull request merged**